### ISSUE TITLE
Suggestion:  link third-party API to classloader config info?

### ISSUE THREAD

**Issue by scottkurz** [OpenLiberty/open-liberty#29607]
We had a recent Q&A, https://stackoverflow.com/questions/77537369/classcastexception-using-eclipselink-descriptorcustomizer-api-in-liberty/77540106 re: configuring 3rd-party API access, for EclipseLink (JPA).

The overall topic info is great... but it kind of assumes that you discover the info from a more general, conceptual level.

OTOH, say I search the info by googling: `liberty org.eclipse.persistence.core.descriptors`  ("bottom-up"?) ... 

I'll find the feature page, and great... I see the entry under:
 > https://openliberty.io/docs/latest/reference/feature/persistence-3.1.html#_third_party_api_packages_provided_by_this_feature

What would be nice though, perhaps, is some kind of link back to the info at:
https://openliberty.io/docs/latest/class-loader-library-config.html 
explaining how you need to configure something like:

```xml
<application id="testApp" name="testApp" type="war" location="testApp.war">
    <classloader apiTypeVisibility="+third-party" />
</application>
```

for this feature and others, like Faces: https://openliberty.io/docs/latest/reference/feature/faces-4.0.html#_third_party_api_packages_provided_by_this_feature

This suggestion was raised in an internal discussion on Slack.

I don't know where/how exactly but let me stop there and see what you think before making a more concrete suggestion.

Thanks

**Comment by dmuelle** [comment]
Hi Scott- I'm glad the 3rd party api info on the  classloader page is helpful, and I think linking from the feature pages makes sense. 

It would require some design work to implement since that section of the feature pages is generated from the feature metatype. That said, seems like we could add a sentence just under the "Third-party API packages provided by this feature" heading that points to the relevant section on the classloader page. I dont know what's involved in updating the generator logic or what the concerns might be.

I dont think we have anything else in the generated doc that links to static pages, but that doesn't necessarily mean we can't or shouldn't. @cbridgha - do you have any insight/feedback on whether this would be feasible?

**Comment by dmuelle** [comment]
I can also bring this up at the next OL website meeting to get feedback from @NottyCode and @mbroz2

**Comment by mbroz2** [comment]
I agree with brining this up for discussion, but in the meantime, here's my thought:
What if we had the generator add something like (similar to what we do for examples and descriptions)
```
include::./[feature]/third-partyAPIs.adoc[opts=optional]
include::./[feature-version]/third-partyAPIs.adoc[opts=optional]
````
after the
`== Third-party API packages provided by this feature`

And then the [feature]/third-partyAPIs.adoc would reference another file (`include::./third-partyAPIs.adoc[opts=optional]`) that would pull in the generic statement that we'd want to add regarding the classloader content we'd want to include.  I'm expecting that in most cases, the single default statement would suffice, but this would allow us to customize it per feature (and per feature version) if the need arose.

If we're pretty sure a single statement would suffice for all features that provide third-party APIs, then we could just make the include a generic one:
`include::./third-partyAPIs.adoc[opts=optional]`

This  avoids placing any such static content (and associated links that can grow stale) in the generated docs.

**Comment by dmuelle** [comment]
moving this to the main OL repo as it's dependent on development rather than docs
